src/browserlib/extract-algorithms.mjs

```diff
@@ -494,24 +494,28 @@ function serializeSteps(root) {
           }
           return {
             'case': getTextContent(option),
-            steps: serializeSteps(dd)
+            steps: serializeSteps(dd, level + 1)
           };
         })
       }
     ]
   }
   else if (root.nodeName === 'OL') {
-    return [...root.querySelectorAll('& > li')].map(serializeStep);
+    return [...root.querySelectorAll('& > li')].map(li => serializeStep(li, level +1));
   }
   else {
-    return [serializeStep(root)];
+    if (level === 0) {
+      return [];
+    } else {
+      return [serializeStep(root, level + 1)];
+    }
   }
 }
 
 /**
  * Serialize an algorithm step
  */
-function serializeStep(li) {
+function serializeStep(li, level) {
```

Comment:
The level parameter is meant for later usage and consistency? Or is there something missing?

Reply:
mostly for consistency, with the idea that it would likely be useful later; but I'm happy to remove it if you feel that's premature

Reply:
Re-reading this, I realized the switch case serialization was convoluted. The case is now serialized directly at the `case` level.

I dropped the `level` parameter everywhere, because it was no longer needed. We can re-introduce it as needed if the level starts to be relevant.